src/main/scala/ai/privado/rulevalidator/YamlFileValidator.scala

```diff
@@ -0,0 +1,195 @@
+package ai.privado.rulevalidator
+
+import ai.privado.entrypoint.CommandConstants
+import ai.privado.model.CatLevelOne
+import better.files.File
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.networknt.schema.{JsonSchema, JsonSchemaFactory, SpecVersion, ValidationMessage}
+import org.slf4j.LoggerFactory
+
+import java.util
+import scala.jdk.CollectionConverters.SetHasAsScala
+import scala.io.Source
+
+
+object YamlFileValidator {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  private val PRETTY_LINE_SEPARATOR = "-"*100
+
+  private val SCHEMA_DIR_PATH = "/ai/privado/rulevalidator/schema/"
+  private val JSON_SCHEMA_VERSION = SpecVersion.VersionFlag.V7
+  private val RULES_FOLDER_IN_CONFIG_DIR = "rules"
+
+  private val SOURCES     = Source.fromInputStream(getClass.getResourceAsStream(s"${SCHEMA_DIR_PATH}sources.json")).mkString
+  private val SINKS       = Source.fromInputStream(getClass.getResourceAsStream(s"${SCHEMA_DIR_PATH}sinks.json")).mkString
+  private val POLICIES    = Source.fromInputStream(getClass.getResourceAsStream(s"${SCHEMA_DIR_PATH}policies.json")).mkString
+  private val THREATS     = Source.fromInputStream(getClass.getResourceAsStream(s"${SCHEMA_DIR_PATH}threats.json")).mkString
+  private val COLLECTIONS = Source.fromInputStream(getClass.getResourceAsStream(s"${SCHEMA_DIR_PATH}collections.json")).mkString
+
+  val mapper = new ObjectMapper(new YAMLFactory())
+
+  val factory: JsonSchemaFactory = JsonSchemaFactory
+    .builder(
+      JsonSchemaFactory
+        .getInstance(JSON_SCHEMA_VERSION)
+    ).objectMapper(mapper)
+    .build()
+
+  implicit def betterFilesExtended(file: File): BetterFileUtil = new BetterFileUtil(file)
+  implicit def jsonSchemaExtended(jsonSchema: JsonSchema): JsonSchemaExtension = new JsonSchemaExtension(jsonSchema)
+
+  /**
+    * Iterate recursively inside a directory path, filter for ".yaml" and ".yml" files
+    * Identify corresponding schema file for each file and validate.
+    * Filter all files to collect files with Validation Errors
+    * @return an Iterator on ai.privado.rulevalidator.ValidationFailure objects
+    */
+  def validateDirectory(dir: File): Iterator[ValidationFailure] = {
+
+    val validationErrors: Iterator[ValidationFailure] = dir
+      .listRecursively
+      .filter(
+        subDir =>
+          subDir
+            .extension(toLowerCase=true)
+            .toString
+            .contains(".yaml")
+            ||
+            subDir
+              .extension(toLowerCase=true)
+              .toString
+              .contains(".yml")
```

Comment:
A shorter way to write this can be (non-blocker)
```.filter(f => f.extension(toLowerCase=true) == Some(".yaml") || f.extension(toLowerCase=true) == Some(".yml"))```